utilities/infra.py

```diff
@@ -583,10 +584,31 @@ def verify_no_failed_pods(
         isvc (InferenceService): InferenceService object
         runtime_name (str): ServingRuntime name
         timeout (int): Time to wait for the pod.
+        timeout_wait_for_pods (int): Time to wait for the pods.
+
     Raises:
-            FailedPodsError: If any pod is in failed state
+        FailedPodsError: If any pod is in failed state
+        TimeoutExpiredError: If no pods were created within `timeout_wait_for_pods` seconds.
 
     """
+    LOGGER.info("Wait for pods creation")
+
+    try:
```

Comment:
This is a great candidate for retry decorator :)

Reply:
this pr is marked as  `wip`, please do not review

Reply:
done